File: iOS/DoraemonKitDemo/DoraemonKitDemo/DemoVC/MemoryLeak/DoraemonDemoMemoryLeakModel.m
//
//  DoraemonDemoMemoryLeakModel.m
//  DoraemonKitDemo
//
//  Created by didi on 2019/10/6.
//  Copyright © 2019 yixiang. All rights reserved.
//

#import "DoraemonDemoMemoryLeakModel.h"

@interface DoraemonDemoMemoryLeakModel()

@property (nonatomic, copy) DoraemonDemoMemoryLeakModelBlock block;

@end

@implementation DoraemonDemoMemoryLeakModel

- (void)addBlock:(DoraemonDemoMemoryLeakModelBlock)block{
    self.block = block;
}

- (void)install{
    self.block();
}

@end

File: iOS/Swift/DoKitSwiftDemo/DoKitSwiftDemo/MemoryLeak/DoraemonDemoMemoryLeakModel.swift
//
//  DoraemonDemoMemoryLeakModel.swift
//  DoKitSwiftDemo
//
//  Created by didi on 2020/5/19.
//  Copyright © 2020 didi. All rights reserved.
//

import UIKit

class DoraemonDemoMemoryLeakModel: NSObject {
    var closure: (()->Void)?
    
    func callClosure() {
        closure?()
    }
    
    deinit {
        print("model deinit")
    }

}
